src/plugins/explore/public/components/visualizations/style_panel/standard_axes_options.tsx

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiPanel,
+  EuiTitle,
+  EuiFormRow,
+  EuiSelect,
+  EuiSwitch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { StandardAxes, Positions } from '../types';
+import { DebouncedTruncateField, DebouncedText } from './utils';
+
+interface AllAxesOptionsProps {
+  standardAxes: StandardAxes[];
+  onStandardAxesChange: (categoryAxes: StandardAxes[]) => void;
+  disableGrid: boolean;
+  onChangeSwitchAxes: (categoryAxes: StandardAxes[]) => void;
+}
+
+interface AxesOptionsProps {
+  standardAxes: StandardAxes[];
+  onStandardAxesChange: (categoryAxes: StandardAxes[]) => void;
+  disableGrid: boolean;
+}
+
+const getAxisDisplayTitle = (axis: StandardAxes) => {
+  if (axis.title?.text && axis.title.text.trim() !== '') {
+    return axis.title.text;
+  }
+  return axis.field?.default.name || '';
+};
+
+export const StandardAxesOptions: React.FC<AxesOptionsProps> = ({
+  standardAxes,
+  onStandardAxesChange,
+  disableGrid,
+}) => {
+  const updateAxis = (index: number, updates: Partial<StandardAxes>) => {
+    const updatedAxes = [...standardAxes];
+    updatedAxes[index] = {
+      ...updatedAxes[index],
+      ...updates,
+    };
```

Comment:
Do we have to shallow copy the entire array and then do mutation? Can we do something like:
```
const updatedAxes = standardAxes.map((axis, i) =>
  i === index ? { ...axis, ...updates } : axis
);
```